test/sql/types/union/union_value.test

```diff
@@ -0,0 +1,43 @@
+# name: test/sql/types/union/union_value.test
+# description: Test union_value function
+# group: [union]
+
```

Comment:
Can we have some more tests:

* Persistent storage of unions
* Default values in UNION tables
* Unions with multiple structs
* Unions with too many tags
* Unions with ambiguous case insensitive names (e.g. "a" and "A")
* test_all_types for the next PR :)